pkg/auth/jwks/keyset.go

```diff
@@ -14,5 +16,5 @@ var (
 )
 
 type KeySet interface {
-	VerifySignature(ctx context.Context, jws string) (payload []byte, err error)
+	VerifySignature(ctx context.Context, jws string, permittedSignatureAlgorithms []jose.SignatureAlgorithm) (payload []byte, err error)
```

Comment:
I'm not sure the algorithms belong as part of this interface. I think it might be better if the KeySet impl was created with permitted algorithms instead. Also I don't think it's a good idea to bake jose types into our interfaces.

Reply:
At the very least we should either specify this as a `[]string` or our own signature type. `jose.SignatureAlgorithm` is just a string underneath anyway.

Reply:
> I don't think it's a good idea to bake jose types into our interfaces.

I agree. I was conscious of not introducing the jose package in any package that didn't previously use it and also, trying to mitigate circular dependencies.

I didn't think introducing it in the interface was a bad idea as the package already included go-jose. I didn't think about keeping the interface clean though 🤦 

Should be a lot better now. Thanks!